ee release with new python measure added

DavidGoldwasser opened the pull request:
No changes to existing measure, only new measure. This measure uses custom libraries and also an API call. I tested on my machine. More work needed for CI system in future and to use with OS app. 

This measure will issue warning if someone uses it but doesn't have API key

prsh5175 committed:
Initial commit. Simply copied a measure as a starting point.

DavidGoldwasser committed:
moving measure over from https://github.com/NREL/openstudio-common-measures-gem

Will be adding addition ee window measure(s) in addition to this one. Will added ability to add embodied carbon.

prsh5175 committed:
Merge branch 'SCOPE-measures' into replace-windows-EC

prsh5175 committed:
Synced with latest parent branch.

DavidGoldwasser committed:
Updating measure.rb to contain description and modeler_description

Now when measure is updated with `openstudio measure -u` updates to description will get pushed to measure.xml and to readme.md

DavidGoldwasser committed:
cleaning up modeler_description and updating tests to save show output and save models

DavidGoldwasser committed:
Adding in a place holder for EC3 API call

Commit:
update the ReplaceExteriorWindowConstruction_Test.rb with EC3 API call

Commit:
API Call edits

prsh5175 committed:
Merge branch 'JY_SCOPE-measures' into replace-windows-EC

prsh5175 committed:
Added minor comments and cleanup.

Commit:
edits

Commit:
change EPD location to US and Canada

prsh5175 committed:
Measure name, description etc. updated and packaged EC3_lookup in a separate helper file.

prsh5175 committed:
Test design document added.

Commit:
Updated version

Commit:
edits on unit conversion

Commit:
calculate GWP mean

prsh5175 committed:
Test commit.

prsh5175 committed:
Test commit

prsh5175 committed:
Initial attempt to add IGU frame gwp (needs bugfixes).

Commit:
bug to be fixed with configparser, unable to read url from config.ini

prsh5175 committed:
Temporary commit.

prsh5175 committed:
Merge branch 'replace-windows-EC' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

prsh5175 committed:
Measure name typo correction and gitignored __pycache__

prsh5175 committed:
Added a reporting line and test updates.

Commit:
translate ruby measure to python in ReplaceExteriorWindowConstruction

Commit:
Merge branch 'replace-windows-EC' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

DavidGoldwasser committed:
removing measure.rb from python measure

DavidGoldwasser committed:
adding some logic to measure to step through subsurfaces

prsh5175 committed:
Python measure and tests updated.

prsh5175 committed:
gitignore, measure.py updated

prsh5175 committed:
Commit for debug.

prsh5175 committed:
perimeter now computed from model vs user argument. Both measure and test file updated. Additional temporary file created to list out all model attributes for faster search.

prsh5175 committed:
Reduced the user-defined arguments.

prsh5175 committed:
layered construction now working.

Commit:
url address extracted using configparser is causing error, can only extract API_TOKEN from config.ini

Commit:
API token and url address extracted using configparser is causing error, temporarily stopped using configparser

Commit:
Revisions: Line 151-178 in measure.py for WindowEnhancement, needs Prateek to review

prsh5175 committed:
Getting rid of some unused lines.

prsh5175 committed:
Minor updates, comments updated, redundant lines deleted. 'total_embodied_carbon' changed to 'frame_embodied_carbon' since IGU calculations are still remaining (added corresponding comments).

prsh5175 committed:
Test commit to check inter-operability with sub-branch.

Commit:
new edits in EC3_lookup.py (new function for generating url) and measure.py for window enhancement (new choice arguments and get min/max/mean gwp values for embodeid carbon calculation)

Commit:
connect generate_url with user arguments; add median gwp option; unable to include industry EPD, need help from EC3

Commit:
saved api token to onedrive folder; code for generating EPD data of igu and window frame is ready

Commit:
consider product lifetime when calculating total embodied carbon; revise lines 371-380 later with the help of Prateek

Commit:
Add notations for subsurface looping; Add edits for window material's volume calcualtion

Commit:
Error free version that can calculate volume of window frame and igu

Commit:
Error free version that can calculate total embodied carbon of window construction

prsh5175 committed:
tests updated and config.ini working for EC3_lookup

prsh5175 committed:
Merge branch 'replace-windows-EC' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

prsh5175 committed:
`apply_measure.py` portion added to `test_window_enhancement.py`

prsh5175 committed:
Dummy additional_properties object added.

prsh5175 committed:
'del model' line added at the end of apply_measure.py for memory leak management.

prsh5175 committed:
Saving the modified model to check that additionalProperties object is added to the OSM.

Commit:
Add FrameAndDivider to example model for window frame calculations; Restructure looping to assign additional properties: embodied carbon of window frame is assigned to "Window Property Frame And Divider 1", embodeid carbon of igu is assigned to "Exterior Window"

Commit:
Add arguments: api_key, epd_type; edit functions: parse_product_epd and parse_industrial_epd

Commit:
Merge branch 'develop' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

Commit:
Increase example model complexity; Reorganize EC3_lookup; Revise window frame calculations, assign embodied carbon to corresponding subsurface object

Commit:
minor revisions and clean up

prsh5175 committed:
Deleted non essential file.

Commit:
handle triple pane, include gap layer thickness for igu calculation

Commit:
Merge branch 'replace-windows-EC' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

prsh5175 committed:
Minor debug.

Commit:
deleted 'single_value' option as it is not necessary; revised keyword for precast concrete EPD search

Commit:
Merge branch 'replace-windows-EC' of https://github.com/NREL/openstudio-ee-gem into replace-windows-EC

prsh5175 committed:
Two example models created with slightly different window properties, measures applied to observe delta EC, ran both models for delta OC

prsh5175 committed:
Re-run of both models.

DavidGoldwasser committed:
Merge pull request #70 from NREL/3.10.0_prep

3.10.0 prep

DavidGoldwasser committed:
removing inadvertent measure

DavidGoldwasser committed:
progress update. Tests not passing, fix or merge_measure to tests

DavidGoldwasser committed:
2 tests running, update other tests and then merge

DavidGoldwasser committed:
Enabled unit tests removed apply_measure

need to add additional tests when epd_type is Indusstry

DavidGoldwasser committed:
do not add replace exterior window construction to this repo yet

It is still on common measure. When Python version with EC capability is ready we can move it to this repo.

DavidGoldwasser committed:
update version readme and change log to get ready for new release

DavidGoldwasser committed:
Merge branch 'develop' into window_enhancement

DavidGoldwasser committed:
Merge pull request #78 from NREL/window_enhancement

Window enhancement ec measure. CI is passing because this is a python measure. Test run locally for me when I have python libraries and EC3 API key.

kflemin commented on lib/measures/window_enhancement/LICENSE.md:
```diff
@@ -0,0 +1 @@
+Insert your license here
```
this doesn't seem right?

kflemin commented on lib/measures/window_enhancement/measure.xml:
```diff
@@ -0,0 +1,146 @@
+      <checksum>26458C20</checksum>
+    </file>
+    <file>
+      <filename>__pycache__/EC3_lookup.cpython-39.pyc</filename>
```
i think you want to remove the pycache files here no?

DavidGoldwasser committed:
hand deleted __pycache__ from measure.xml by hand. running update didn't work

We have had issues with file reference not always being cleaned up. Need to investigate this more later.

DavidGoldwasser committed:
updated license for window enhance python measure by hand

rake task in ext gem for openstudoi:update_copyright didn't catch this

DavidGoldwasser committed:
Merge pull request #81 from NREL/win_enh_cleanup

Win enh cleanup

anchapin committed:
Improve GitHub Actions workflow reliability and error handling

This commit enhances the test-with-openstudio workflow with better error
handling, simplified logic, and improved debugging capabilities.

Key Changes:
- Update Docker image to nrel/openstudio:develop for latest features
- Add proper locale configuration (locales package + locale-gen)
- Fix git safe.directory configuration (was hardcoded for wrong repo)
- Simplify test execution by removing redundant double-run approach
- Add comprehensive error handling with GitHub Actions annotations
- Improve S3 sync with explicit error checking and diagnostics
- Fix environment variable handling for artifact uploads

Technical Details:
- Replaced error suppression (set +e) with proper error handling
- Added ::error:: annotations for better GitHub Actions integration
- Added success indicators (✅) for better workflow visibility
- Fixed locale generation to match container LANG setting
- Streamlined bundle operations to avoid unnecessary complexity

Testing:
- Created test-workflow-locally.sh for local validation
- All critical paths now have explicit error handling
- S3 sync includes directory validation and detailed error messages

Benefits:
- Faster workflow execution (removed redundant operations)
- Clearer failure identification and debugging
- More reliable S3 deployments
- Better artifact handling
- Enhanced workflow logs with diagnostic information

Fixes issues with:
- Git ownership errors in Docker container
- Locale configuration mismatches
- Silent failures in test execution
- Unclear error messages for debugging

anchapin committed:
Update OpenStudio workflow to use stable image and add upgrade check; remove Jenkinsfile

anchapin committed:
Refactor GitHub Actions workflow to use OpenStudio develop image, streamline test execution, and enhance environment setup

anchapin committed:
Update Docker Hub authentication in GitHub Actions workflow to use environment variables

anchapin committed:
Fix Docker Hub authentication by correcting secret variable usage in GitHub Actions workflow

anchapin committed:
Update Gemfile to use secure HTTPS source for RubyGems and enhance GitHub Actions workflow with improved dependency caching and error handling

anchapin committed:
Add .coverage file, update .gitignore for environment and secrets, and fix measure.xml paths

anchapin committed:
Merge pull request #83 from NREL/remove_jenkins

Improve GitHub Actions workflow reliability and error handling

anchapin committed:
Update measures and tests for fan assist night ventilation, NZE HVAC, and replace water heater with thermal storage chilled water

- Updated version IDs and modified dates in measure.xml files for fan_assist_night_ventilation and nze_hvac.
- Changed checksums for measure.rb and test files in fan_assist_night_ventilation and nze_hvac.
- Refactored test files to improve readability and maintainability by removing unnecessary conditionals and using guard clauses.
- Enhanced the NZE HVAC measure to improve HVAC system partitioning logic and error handling.
- Updated the replace_water_heater_mixed_with_thermal_storage_chilled_water measure to streamline component replacement logic.
- Added development dependencies in openstudio-ee.gemspec for improved development workflow.

anchapin committed:
Fix duplicate condition check in NZEHVAC_Test.rb

Addresses GitHub Copilot review feedback by removing the redundant inner condition check that created unreachable code.

anchapin committed:
Merge pull request #84 from NREL/release_prep_0-12-4

Update to v6 of rubocop styles and update license
